### PR TITLE
RUN-1233 - Assign logged in user to jobs where possible

### PR DESCRIPTION
**Pull request by 85danf**

### CODE REVIEW

File: cmd/job/submit/submit.go [hunk 1]
```diff
@@ -292,6 +287,19 @@ func (submitArgs *submitArgs) setCommonRun(cmd *cobra.Command, args []string, ku
...
 
+func assignUser(submitArgs *submitArgs) {
+	// If its not zero then it was passed as flag by the user
+	if submitArgs.User == "" {
+		if kubeLoginUser, err := auth.GetEmailForCurrentKubeloginToken(); err == nil && kubeLoginUser != "" {
+			// Try to get user from kubelogin cached token
+			submitArgs.User = kubeLoginUser
+		} else if osUser, err := user.Current(); err == nil {
+			// Fallback to OS user
+			submitArgs.User = osUser.Username
```
Comment (yodarshafrir1) [inline cmd/job/submit/submit.go:298]: why change the old logic and not use:
```
        currentUser, err := user.Current()
	if err != nil {
		defaultUser = ""
	} else {
		defaultUser = currentUser.Username
	}
```

Comment (yodarshafrir1) [reply cmd/job/submit/submit.go:298]: what happens on the first submit, when there's no token?

**Review by yodarshafrir1 — CHANGES_REQUESTED** [review]

File: cmd/job/submit/submit.go (hunk 1 above)
Comment (85danf) [reply cmd/job/submit/submit.go:298]: How is the new logic 
```
if osUser, err := user.Current(); err == nil {
			// Fallback to OS user
			submitArgs.User = osUser.Username
```

different from the old logic?

```
     currentUser, err := user.Current()
	if err != nil {
		defaultUser = ""
	} else {
		defaultUser = currentUser.Username
	}
```

The difference is purely semantic.


This logic relies on the permission check being run beforehand (thus populating the token as part of the check itself).
In cases where token auth is not used the user will be the OS user, like it was up until now

Comment (yodarshafrir1) [reply cmd/job/submit/submit.go:298]: sorry i though osUser is taken from somewhere else. ignore this comment

**Review by yodarshafrir1 — APPROVED** [review]

**Pull request merged**